flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -335,15 +371,30 @@ class CodeGenerator(
         j"""
             |  public final void accumulate(
             |    org.apache.flink.types.Row accs,
-            |    org.apache.flink.types.Row input)""".stripMargin
+            |    org.apache.flink.types.Row input) throws Exception""".stripMargin
 
+      val distAggsFlags: Array[Boolean] = distinctAggsFlags.getOrElse(new Array[Boolean](0))    
       val accumulate: String = {
         for (i <- aggs.indices) yield
-          j"""
-             |    ${aggs(i)}.accumulate(
-             |      ((${accTypes(i)}) accs.getField($i)),
-             |      ${parameters(i)});""".stripMargin
-      }.mkString("\n")
+          if (distinctAggsFlags.isDefined && distAggsFlags(i)){
+            j"""
+               |  Long distValCount$i = (Long) distStateList[$i].get(${parameters(i)});
```

Comment:
We need to put the parameters into a tuple before accessing the map state. The tuple should be reused.